api/handlers_e2e_test.go

```diff
@@ -1428,3 +1435,354 @@ func TestLookupMultiInnerLogs(t *testing.T) {
 		})
 	}
 }
+
+// compareAppBoxesAgainstHandler is of type BoxTestComparator
+func compareAppBoxesAgainstHandler(t *testing.T, db *postgres.IndexerDb,
+	appBoxes map[basics.AppIndex]map[string]string, deletedBoxes map[basics.AppIndex]map[string]bool, verifyTotals bool) {
+
+	setupRequest := func(path, paramName, paramValue string) (echo.Context, *ServerImplementation, *httptest.ResponseRecorder) {
+		e := echo.New()
+		req := httptest.NewRequest(http.MethodGet, "/", nil)
+		rec := httptest.NewRecorder()
+		c := e.NewContext(req, rec)
+		c.SetPath(path)
+		c.SetParamNames(paramName)
+		c.SetParamValues(paramValue)
+		api := testServerImplementation(db)
+		return c, api, rec
+	}
+
+	remainingBoxes := map[basics.AppIndex]map[string]string{}
+	numRequests := 0
+	sumOfBoxes := 0
+	sumOfBoxBytes := 0
+
+	caseNum := 1
+	var totalBoxes, totalBoxBytes int
+	for appIdx, boxes := range appBoxes {
+		totalBoxes = 0
+		totalBoxBytes = 0
+
+		remainingBoxes[appIdx] = map[string]string{}
+
+		// compare expected against handler response one box at a time
+		for key, expectedValue := range boxes {
+			msg := fmt.Sprintf("caseNum=%d, appIdx=%d, key=%#v", caseNum, appIdx, key)
+			expectedAppIdx, boxName, err := logic.SplitBoxKey(key)
+			require.NoError(t, err, msg)
+			require.Equal(t, appIdx, expectedAppIdx, msg)
+			numRequests++
+
+			boxDeleted := false
+			if deletedBoxes != nil {
+				if _, ok := deletedBoxes[appIdx][key]; ok {
+					boxDeleted = true
+				}
+			}
+
+			c, api, rec := setupRequest("/v2/applications/:appidx/box/", "appidx", strconv.Itoa(int(appIdx)))
+			prefixedName := fmt.Sprintf("str:%s", boxName)
+			params := generated.LookupApplicationBoxByIDAndNameParams{Name: prefixedName}
+			err = api.LookupApplicationBoxByIDAndName(c, uint64(appIdx), params)
+			require.NoError(t, err, msg)
+			require.Equal(t, http.StatusOK, rec.Code, fmt.Sprintf("msg: %s. unexpected return code, body: %s", msg, rec.Body.String()))
+
+			var resp generated.BoxResponse
+			data := rec.Body.Bytes()
+			err = json.Decode(data, &resp)
+
+			if !boxDeleted {
+				require.NoError(t, err, msg, msg)
+				require.Equal(t, boxName, string(resp.Name), msg)
+				require.Equal(t, expectedValue, string(resp.Value), msg)
+
+				remainingBoxes[appIdx][boxName] = expectedValue
+
+				totalBoxes++
+				totalBoxBytes += len(boxName) + len(expectedValue)
+			} else {
+				require.ErrorContains(t, err, "no rows in result set", msg)
+			}
+		}
+
+		msg := fmt.Sprintf("caseNum=%d, appIdx=%d", caseNum, appIdx)
+
+		expectedBoxes := remainingBoxes[appIdx]
+
+		c, api, rec := setupRequest("/v2/applications/:appidx/boxes", "appidx", strconv.Itoa(int(appIdx)))
+		// TODO: should we add Order to the search params?
+		params := generated.SearchForApplicationBoxesParams{}
+
+		// TODO: also test non-nil Limit, Next
```

Comment:
@tzaffi So that I know how to gauge the PR:  Do you intend to address the highlighted TODO in-scope to the PR?

Reply:
I've been revisiting and resolving these from time to time and hoping that a reviewer would have an obvious answer. I also am waiting for @winder to have a look. I will aim to research the remaining 2 items asap and either resolve or tag someone with an actionable question.

Reply:
Oh... didn't realize this was referencing a line of code... Thanks for reminding me of these TODO's. I will look now as we ought not have any remaining TODO's.

Reply:
Answer:
* Order is currently fixed as Ascending. I will make a note of this in the PR description and a TODO that we should gain agreement on this omission, or on a going-forward strategy.
* `non-nil Limit, next`[ have been tested](https://github.com/algorand/indexer/blob/5f1192e9af72214d39b38de2accf50454d760a4d/api/test_resources/boxes.json#L774).

Reply:
@tzaffi Got it - resolving.